src/main/java/net/sf/jabref/gui/actions/CleanUpAction.java

```diff
@@ -725,4 +740,22 @@ private static void doConvertLaTeX(BibtexEntry entry, NamedCompound ce) {
         }
     }
 
+    /*
+     * Format dates correctly (yyyy-mm-dd or yyyy-mm)
+     */
+    private static void doCleanUpDate(BibtexEntry entry, NamedCompound ce) {
+        String oldValue = entry.getField("date");
+        // not set
+        if (oldValue == null) {
+            return;
+        }
+        // run formatter
+        new DatesCleanup(entry).cleanup();
```

Comment:
The dates cleanup class is not required. Have a look at the doConvertUnits method. It should simply call the formatter which formats the string.